borbskeeper/Features/Profile/ProfileViewController.m

```diff
@@ -0,0 +1,162 @@
+@property (nonatomic, strong) UIRefreshControl *refreshControl;
+
+@end
+
+@implementation ProfileViewController
+
+static NSString *const USER_PROF_PIC_KEY = @"profilePicture";
+static NSString *const COMPLETE_TASK_TABLE_VIEW_CELL_ID = @"CompletedTaskCell";
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.completedTasksListInfiniteScrollView.infiniteScrollDelegate = self;
+    [self setupProfile];
+    [self fetchData];
+    [self refreshCompleteTaskList];
+}
+
+- (void)setupProfile{
+    NSString *userName = User.currentUser.username;
+    self.userName.text = userName;
+    [self loadUserProfilePicture];
+}
+
+-(void)loadUserProfilePicture{
+    PFFileObject *PFObjectProfileImage = [User currentUser][USER_PROF_PIC_KEY];
+    NSURL *profileImageURL = [NSURL URLWithString:PFObjectProfileImage.url];
+    self.profilePicture.image = nil;
+    [self.profilePicture setImageWithURL:profileImageURL];
+}
+
+- (UIImage *)resizeImage:(UIImage *)image withSize:(CGSize)size {
+    UIImageView *resizeImageView = [[UIImageView alloc] initWithFrame:CGRectMake(0, 0, size.width, size.height)];
+    resizeImageView.contentMode = UIViewContentModeScaleAspectFill;
+    resizeImageView.image = image;
+    
+    UIGraphicsBeginImageContext(size);
+    [resizeImageView.layer renderInContext:UIGraphicsGetCurrentContext()];
+    UIImage *newImage = UIGraphicsGetImageFromCurrentImageContext();
+    UIGraphicsEndImageContext();
+    return newImage;
+}
+
+- (void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary<NSString *,id> *)info {
+    self.originalImage = info[UIImagePickerControllerOriginalImage];
+    self.editedImage = [self resizeImage:self.originalImage withSize:CGSizeMake(1000, 1000)];
+    [self.profilePicture setImage:self.editedImage];
+    
+    [User currentUser][USER_PROF_PIC_KEY] = [BorbParseManager getPFFileFromImage:self.editedImage];
+    
+    [BorbParseManager saveUser:[User currentUser] withCompletion:nil];
+    
+    [self dismissViewControllerAnimated:YES completion:nil];
+}
+
+- (IBAction)didTapChangeProfilePicture:(id)sender {
+    UIImagePickerController *imagePickerVC = [UIImagePickerController new];
+    imagePickerVC.delegate = self;
+    imagePickerVC.allowsEditing = YES;
+    
+    if ([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera]) {
+        imagePickerVC.sourceType = UIImagePickerControllerSourceTypeCamera;
+    } else {
+        imagePickerVC.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
+    }
+    [self presentViewController:imagePickerVC animated:YES completion:nil];
+}
+
+- (void)refreshCompleteTaskList{
+    self.refreshControl = [[UIRefreshControl alloc] init];
+    [self.refreshControl addTarget:self action:@selector(fetchData) forControlEvents:UIControlEventValueChanged];
+    [self.completedTasksListInfiniteScrollView.tableView insertSubview:self.refreshControl atIndex:0];
```

Comment:
We should probably let the `InifiniteScrollView` do this for us.

Reply:
Yea, I think so too.